### PR TITLE
Add manage_init_script parameter

### LINKED ISSUE #541: Drop management of Init script
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: All
- Ruby: All
- Distribution: All
- Module version: Since #240

## What are you seeing
The module manages the init script for agent and server. This was done because some custom settings e.g. `configdir` require a change for startup values of the zabbix agent|server process. In my opinion managing the init file is bad because it leads to a lot of problems with e.g. updates and should only be done if there is *no* other option.

## What behaviour did you expect instead
Use the OS's `defaults` file:

Debian -> /etc/default/zabbix-agent
RedHat -> /etc/sysconfig/zabbix-agent
SystemD -> /etc/systemd/system/zabbix-agent.service.d/override.conf

## Any additional information you'd like to impart

Are you interested (and are willing to merge it ;)) in a PR which moves the configuration to the mentioned default files?

### PR DESCRIPTION
**Pull request by baurmatt**
#### Pull Request (PR) description
Adds a parameter `manage_init_script` which makes it possible to disable the Init File management.

#### This Pull Request (PR) fixes the following issues
Fixes #541.

### CODE REVIEW

**Comment by baurmatt** [comment]
I'm pretty sure the failing test isn't related to my code change. In fact, i can't even reproduce this on my local machine oO

File: manifests/agent.pp [hunk 1]
```diff
@@ -177,6 +177,9 @@
 # [*loadmodule*]
 #   Module to load at agent startup.
 #
+# [*manage_init_script*]
```
Comment (alexjfisher) [inline manifests/agent.pp]: `zabbix::startup` manages systemd instead of init scripts on some systems, so this parameter name may not be the best choice.

Comment (alexjfisher) [reply manifests/agent.pp]: works for me.  @bastelfreak - sound ok?

Comment (bastelfreak) [reply manifests/agent.pp]: sounds good to me. @baurmatt can you update it please?

Comment (baurmatt) [reply manifests/agent.pp]: Done. Didn't change the branch name though ;)

**Comment by bastelfreak** [comment]
@alexjfisher can you have a look again?

**Pull request merged**